### PR TITLE
SI-7995 completion imported vars and vals

### PR DESCRIPTION
**Pull request by skyluc**
Imported member vals and vars were always marked inaccessible, even
if referencing them at the location of the completion is valid in code.
The accessible flag is now set accordingly to the accessibility of the getter.

I am working on an equivalent PR for 2.11. The changes are fairly different, as the code has been modularized, and the format of the oracle is different. I'll submit it after this PR is merged.

### CODE REVIEW

**Comment by skyluc** [comment]
@adriaanm @retronym, can somebody tell me what pr-scala-test is testing, and if it looks like a valid failure?

**Comment by retronym** [comment]
`pr-scala-test` runs the full test suite.

It failed at:

```
 [partest] testing: [...]/files/presentation/scope-completion-2                  [FAILED]
  [partest] reload: Completions.scala
  [partest] 
  [partest] askScopeCompletion at Completions.scala(15,2)
  [partest] ================================================================================
  [partest] [response] askScopeCompletion at (15,2)
  [partest] retrieved 10 members
  [partest] [accessible:  true] `class Cc1Completion1.this.Cc1`
  [partest] [accessible:  true] `class Co1test.Completion1.Co1`
  [partest] [accessible:  true] `class Completion1test.Completion1`
  [partest] [accessible:  true] `constructor Completion1()test.Completion1`
  [partest] [accessible:  true] `method fc1=> Int`
  [partest] [accessible:  true] `method fo1=> Int`
  [partest] [accessible:  true] `object Completion1test.Completion1.type`
  [partest] [accessible:  true] `value ctest.Completion1`
  [partest] [accessible:  true] `value vc1Int`
  [partest] [accessible:  true] `value vo1Int`
  [partest] ================================================================================
  [partest] 
  [partest] askScopeCompletion at Completions.scala(29,2)
  [partest] ================================================================================
  [partest] [response] askScopeCompletion at (29,2)
  [partest] retrieved 10 members
  [partest] [accessible:  true] `class Cc1test.Completion1.c.Cc1`
  [partest] [accessible:  true] `class Co1test.Completion1.Co1`
  [partest] [accessible:  true] `class Completion1test.Completion1`
  [partest] [accessible:  true] `constructor Completion1()test.Completion1.type`
  [partest] [accessible:  true] `method fc1=> Int`
  [partest] [accessible:  true] `method fo1=> Int`
  [partest] [accessible:  true] `object Completion1test.Completion1.type`
  [partest] [accessible:  true] `value ctest.Completion1`
  [partest] [accessible:  true] `value vc1Int`
  [partest] [accessible:  true] `value vo1Int`
  [partest] ================================================================================
...
BUILD FAILED
/localhome/jenkins/c/workspace/pr-scala-test/scala/build.xml:1583: Test suite finished with 1 case failing:
/localhome/jenkins/c/workspace/pr-scala-test/scala/test/files/presentation/scope-completion-2 [FAILED]
```

**Comment by retronym** [comment]
This differs from the committed checkfile:

``` diff
-askScopeCompletion at Completions.scala(16,4)
+askScopeCompletion at Completions.scala(15,2)
 ================================================================================
-[response] askScopeCompletion at (16,4)
-retrieved 11 members
+[response] askScopeCompletion at (15,2)
+retrieved 10 members
 [accessible:  true] `class Cc1Completion1.this.Cc1`
 [accessible:  true] `class Co1test.Completion1.Co1`
 [accessible:  true] `class Completion1test.Completion1`
 [accessible:  true] `constructor Completion1()test.Completion1`
 [accessible:  true] `method fc1=> Int`
 [accessible:  true] `method fo1=> Int`
-[accessible:  true] `method test=> Unit`
 [accessible:  true] `object Completion1test.Completion1.type`
 [accessible:  true] `value ctest.Completion1`
 [accessible:  true] `value vc1Int`
 [accessible:  true] `value vo1Int`
 ================================================================================
-
-askScopeCompletion at Completions.scala(32,4)
+askScopeCompletion at Completions.scala(29,2)
 ================================================================================
-[response] askScopeCompletion at (32,4)
-retrieved 11 members
+[response] askScopeCompletion at (29,2)
+retrieved 10 members
 [accessible:  true] `class Cc1test.Completion1.c.Cc1`
 [accessible:  true] `class Co1test.Completion1.Co1`
 [accessible:  true] `class Completion1test.Completion1`
 [accessible:  true] `constructor Completion1()test.Completion1.type`
 [accessible:  true] `method fc1=> Int`
 [accessible:  true] `method fo1=> Int`
-[accessible:  true] `method test=> Unit`
 [accessible:  true] `object Completion1test.Completion1.type`
 [accessible:  true] `value ctest.Completion1`
 [accessible:  true] `value vc1Int`
```

**Comment by skyluc** [comment]
Silly me, I thought partest was run in the CI phase called `*partest*` :smiley: 

The check file is wrong. Fixing it.

File: src/compiler/scala/tools/nsc/interactive/Global.scala [hunk 1]
```diff
@@ -955,7 +955,12 @@ class Global(settings: Settings, _reporter: Reporter, projectName: String = "")
     val enclosing = new Members[ScopeMember]
     def addScopeMember(sym: Symbol, pre: Type, viaImport: Tree) =
       locals.add(sym, pre, false) { (s, st) =>
-        new ScopeMember(s, st, context.isAccessible(s, pre, false), viaImport)
+        // imported val and var are always marked as inaccessible, but they could be accessed through their getters. SI-7995
+        if (s.hasGetter) {
```
Comment (JamesIry) [inline src/compiler/scala/tools/nsc/interactive/Global.scala]: brackets on the one line if but not on the one line else.

File: src/compiler/scala/tools/nsc/interactive/Global.scala [hunk 2]
```diff
@@ -955,7 +955,12 @@ class Global(settings: Settings, _reporter: Reporter, projectName: String = "")
     val enclosing = new Members[ScopeMember]
     def addScopeMember(sym: Symbol, pre: Type, viaImport: Tree) =
       locals.add(sym, pre, false) { (s, st) =>
-        new ScopeMember(s, st, context.isAccessible(s, pre, false), viaImport)
+        // imported val and var are always marked as inaccessible, but they could be accessed through their getters. SI-7995
+        if (s.hasGetter) { 
+          new ScopeMember(s, st, context.isAccessible(s.getter, pre, superAccess = false), viaImport)
```
Comment (dotta) [inline src/compiler/scala/tools/nsc/interactive/Global.scala]: I'm wondering, shouldn't `s.getter` be passed to `ScopeMember` as first argument?

Comment (skyluc) [inline src/compiler/scala/tools/nsc/interactive/Global.scala]: Nope. With `s.getter`, the scope completion would reference the getter method (which is synthetic) and not the var/val.

Comment (dotta) [inline src/compiler/scala/tools/nsc/interactive/Global.scala]: Right. Thanks for explaining.

**Comment by skyluc** [comment]
I fixed the braces inconsistency.

**Comment by adriaanm** [comment]
PLS REBUILD ALL

**Comment by adriaanm** [comment]
LGTM

**Pull request merged**